src/Tools/GenerateDocumentationAndConfigFiles/Program.cs

```diff
@@ -361,6 +361,10 @@ void createAnalyzerDocumentationFile()
                     description = Regex.Replace(description, "(\r?\n)", "$1$1");
                     // Escape generic arguments to ensure they are not considered as HTML elements
                     description = Regex.Replace(description, "(<.+?>)", "\\$1");
+                    // Add angle brackets around links to prevent violating MD034:
+                    // https://github.com/DavidAnson/markdownlint/blob/82cf68023f7dbd2948a65c53fc30482432195de4/doc/Rules.md#md034---bare-url-used
+                    // Regex taken from: https://github.com/DavidAnson/markdownlint/blob/59eaa869fc749e381fe9d53d04812dfc759595c6/helpers/helpers.js#L24
+                    description = Regex.Replace(description, @"(?:http|ftp)s?:\/\/[^\s\]""']*(?:\/|[^\s\]""'\W])", "<$0>");
```

Comment:
You might want to add `?` after the `*` in the regex for better results.

```suggestion
                    description = Regex.Replace(description, @"(?:http|ftp)s?:\/\/[^\s\]""']*?(?:\/|[^\s\]""'\W])", "<$0>");
```

Reply:
@Evangelink I took the regex as-is from markdownlint repository.

Do you have an example where the current regex won't work but the modified will? In that case it would be a possible bug in markdownlint then.

Reply:
It won't be a bug, just a performance hotpoint. This is related to some backtracking, I don't remember where is the doc.

Reply:
@Evangelink I reverted your change here, it caused the closing angle bracket to be in the wrong position.

![image](https://user-images.githubusercontent.com/31348972/103177074-d82ff480-487f-11eb-8add-191887fda05b.png)

Reply:
Oh it makes sense based on the rest of the regex, my bad. To use this non-greedy version we would need to tweak the rest. I guess that's fine as-is.